lalrpop/src/normalize/tyinfer/mod.rs

```diff
@@ -273,7 +273,8 @@ impl<'grammar> TypeInferencer<'grammar> {
             SymbolKind::Name(_, ref s) => self.symbol_type(&s.kind),
 
             SymbolKind::Repeat(..) | SymbolKind::Expr(..) | SymbolKind::Macro(..) |
-            SymbolKind::AmbiguousId(..) | SymbolKind::Lookahead | SymbolKind::Lookbehind => {
+            SymbolKind::AmbiguousId(..) | SymbolKind::Lookahead | SymbolKind::Lookbehind |
+            SymbolKind::Error => {
```

Comment:
Wait, I don't think `SymbolKind::Error` belongs here -- it doesn't get macro-expanded away?

I think this would generate an error if you had something like

```rust
Foo = error;
```

which would cause us to try and infer the type of `Foo`.